src/store.rs

```diff
@@ -1473,6 +1473,9 @@ pub fn store_git_tree(tree_buf: &[u8], reference: Option<TreeId>) -> TreeId {
         let mut ref_tree = None;
         if let Some(reference) = reference {
             oe = get_object_entry(&reference.into());
+
+            // c_char can either be u8 or i8 depending on each platform
+            #[allow(clippy::unnecessary_cast)]
```

Comment:
Had to put this on the block as putting it on `ref_tree = ...` causes ["attributes on expressions are experimental"](https://github.com/rust-lang/rust/issues/15701) error.